docs/observation/README.md

```diff
@@ -0,0 +1,99 @@
+  - update the `helix-bot-config` with the `trigger-namespace` and `trigger-token`
+- the `helix-services/observation` also forwards events received via `POST /feed/:owner/:repo/:ref` to the respective trigger.
+
+### Authorization
+
+Only events that either include `trigger-token` or are properly signed are forwarded to the trigger.
+- if the events contain a `x-trigger-token` header:
+  - the `helix-services/observation` action reads (and caches) the `trigger-token` annotation form the respective trigger
+  - if the trigger tokens don't match, it returns with a 403.
+
+- if the events contain a `x-bot-signature` header:
+  - the `helix-services/observation` action checks that the payload is signed with the bot's private key.
+  - if the signature is invalid, it returns a 403.
+
+**Note**:
+
+currently, the helix-bot is the only client that uses the `helix-services/observation` action directly, so we could
+drop the `trigger-token` feature. but for testing and potential other clients, it is still useful to have.
+
+### helix project (not helix-pages)
+
+#### bot setup
+similar to the way the purge configuration is updated during `hlx publish`, the step also:
+- ensures there's a trigger `{owner}--{repo}--{ref}` for all content repositories where a bot is installed
+- if not, create a trigger using the ``helix-services/observation` feed.
+
+#### additional content change listeners
+In order to register additional content change listeners, for example for onedrive, the setup is similar.
+
+#### onedrive
+for onedrive, each content repository that contains a `fstab.yaml` with mountpoints with a onedrive root need a _onedrive subscription_ that eventually invokes a change listener.
+- during `hlx publish` all `{repository, driveRoot}` tuples are used to update the subscriptions.
+- managing the subscriptions should probably happen in a separate action, but invoked by the `observation` feed.
+- each subscription needs to use the `onedrive-change-listener` action (webhook) in the respective customer namespace, so that the change listener can fire the trigger (without extra authentication).
+
+#### dispatch
+
+when the helix-bot receives a repository change event, it will
+- read the `trigger-namespace` and from the `helix-bot-config.yaml`
+- sign the payload with its private-key
+- `POST` the event information to the respective `observation` action eg:
+   `https://adobeioruntime.net/api/v1/web/helix-test/observation/feed/myowner/myrepo/master`
+
+
+### helix-pages project
+
+The main use of observation for a helix-pages project is the updating of the index when content is modified. Since the helix-pages projects don't have their own OpenWhisk namespace, all triggers are installed in the `helix-pages` namespace.
+
+setting up a trigger and creating the algolia index could be done automatically when the helix-bot is added to the repository. however, it is not clear yet how much power the bot should have here.
```

Comment:
Would it make sense to use the visitor data in Google BigQuery to trigger the indexing for Helix Pages repositories that don't have the bot installed? We'd run a cron job that queries the traffic logs for Helix Pages to find pages that have been modified since the last query run≥

Reply:
it probably depends on how many indexes you want to create in algolia. IMO, only customers that really need the query index, should also get one.

Reply:
In that case installing the bot is the best way of signaling that you want the index.